client/core/core.go

```diff
@@ -6670,3 +6672,275 @@ func (c *Core) WalletLogFilePath(assetID uint32) (string, error) {
+	csvWriter.UseCRLF = runtime.GOOS == "windows"
+	err = csvWriter.Write([]string{
+		"Host",
+		"Order ID",
+		"Base",
+		"Quote",
+		"Base Quantity",
+		"Order Rate",
+		"Actual Rate",
+		"Base Fees",
+		"Quote Fees",
+		"Type",
+		"Side",
+		"Time in Force",
+		"Status",
+		"TargetOrderID",
+		"Filled (%)",
+		"Settled (%)",
+		"Time",
+	})
+	if err != nil {
+		ordersFile.Close()
+		return nil, nil, fmt.Errorf("error writing CSV: %v", err)
+	}
+	csvWriter.Flush()
+	err = csvWriter.Error()
+	if err != nil {
+		ordersFile.Close()
+		return nil, nil, fmt.Errorf("error writing CSV: %v", err)
+	}
+	return func(ord *db.MetaOrder) error {
+		cord := coreOrderFromTrade(ord.Order, ord.MetaData)
+
+		baseUnitInfo, err := asset.UnitInfo(cord.BaseID)
+		if err != nil {
+			return fmt.Errorf("unable to get base unit info for %v: %v", cord.BaseSymbol, err)
+		}
+		quoteUnitInfo, err := asset.UnitInfo(cord.QuoteID)
+		if err != nil {
+			return fmt.Errorf("unable to get quote unit info for %v: %v", cord.QuoteSymbol, err)
+		}
+		ordReader := &OrderReader{
+			Order:         cord,
+			BaseUnitInfo:  baseUnitInfo,
+			QuoteUnitInfo: quoteUnitInfo,
+		}
+
+		timestamp := encode.UnixTimeMilli(int64(cord.Stamp)).Local().Format(time.RFC3339Nano)
+		err = csvWriter.Write([]string{
+			cord.Host,                     // Host
+			ord.Order.ID().String(),       // Order ID
+			cord.BaseSymbol,               // Base
+			cord.QuoteSymbol,              // Quote
+			ordReader.BaseQtyString(),     // Base Quantity
+			ordReader.SimpleRateString(),  // Order Rate
+			ordReader.AverageRateString(), // Actual Rate
+			ordReader.BaseAssetFees(),     // Base Fees
+			ordReader.QuoteAssetFees(),    // Quote Fees
+			ordReader.Type.String(),       // Type
+			ordReader.SideString(),        // Side
+			cord.TimeInForce.String(),     // Time in Force
+			ordReader.StatusString(),      // Status
+			cord.TargetOrderID.String(),   // Target Order ID
+			ordReader.FilledPercent(),     // Filled
+			ordReader.SettledPercent(),    // Settled
+			timestamp,                     // Time
+		})
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+		csvWriter.Flush()
+		err = csvWriter.Error()
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+		return nil
+	}, ordersFile.Close, nil
+}
+
+func deleteMatchFn(matchesFileStr string) (perMatchFn func(*db.MetaMatch, bool) error, cleanUpFn func() error, err error) {
+	matchesFile, err := createFile(matchesFileStr)
+	if err != nil {
+		return nil, nil, fmt.Errorf("problem opening orders file: %v", err)
+	}
+	csvWriter := csv.NewWriter(matchesFile)
+	csvWriter.UseCRLF = runtime.GOOS == "windows"
+
+	err = csvWriter.Write([]string{
+		"Host",
+		"Base",
+		"Quote",
+		"Match ID",
+		"Order ID",
+		"Quantity",
+		"Rate",
+		"Swap Fee Rate",
+		"Swap Address",
+		"Status",
+		"Side",
+		"Secret Hash",
+		"Secret",
+		"Contract Data",
+		"Counter Contract",
+		"Counter TxData",
```

Comment:
After testing a bit with some big db files, I think we should omit the counter tx data.  It's just the raw byte of the counterparty's contract txn, which can be obtained elsewhere.

I'm on the fence about the "Contract Data" and "Counter Contract" too.

Reply:
Can remove...

Reply:
Removed.